packages/web-sdk/src/instrumentations/webVitalsWithAttribution/instrumentation.ts

```diff
@@ -0,0 +1,128 @@
+import { onCLS, onFCP, onFID, onINP, onLCP, onTTFB } from 'web-vitals/attribution'
+import type { Metric } from 'web-vitals/attribution'
+
+import { BaseInstrumentation, VERSION } from '@grafana/faro-core';
+import type  { MeasurementEvent, PushMeasurementOptions } from '@grafana/faro-core'
+
+type Values = MeasurementEvent['values']
+type Context = Required<PushMeasurementOptions>['context']
+
+export class WebVitalsWithAttributionInstrumentation extends BaseInstrumentation {
+  readonly name = '@grafana/faro-web-sdk:instrumentation-web-vitals-with-attribution'
+  readonly version = VERSION
+
+  initialize(): void {
+    this.logDebug('Initializing');
+    this.measureCLS()
+    this.measureFCP()
+    this.measureFID()
+    this.measureINP()
+    this.measureLCP()
+    this.measureTTFB()
+  }
+
+  private measureCLS(): void {
+    onCLS((metric) => {
+      const values = this.buildInitialValues(metric)
+      this.addIfPresent(values, 'largest_shift_entry', metric.attribution.largestShiftEntry?.value)
```

Comment:
Another option would be to create a function that goes through all the keys of the `attribution` object and saves them in `values` or `context` according to their `type of` but these values are not always a primitive value so I opted for this slightly more verbose implementation